sdk/core/Azure.Core/src/DynamicData/DynamicData.cs

```diff
@@ -31,11 +31,15 @@ public sealed partial class DynamicData : IDisposable
 
         private MutableJsonElement _element;
         private DynamicDataOptions _options;
+        private JsonSerializerOptions _serializerOptions;
 
         internal DynamicData(MutableJsonElement element, DynamicDataOptions options = default)
         {
             _element = element;
             _options = options;
+            _serializerOptions = options.NameMapping == DynamicDataNameMapping.None ?
+                new JsonSerializerOptions() :
+                new JsonSerializerOptions() { PropertyNameCaseInsensitive = true };
```

Comment:
Given the other two current options, this doesn't feel too "loose"? Granted, I don't know how else you'd qualify this in the enum. Probably fine. Just wondering.

Reply:
I changed it to use PropertyNamingPolicy instead -- this should more closely match the DynamicDataOptions value.